docs/content/guides/dagster/validating_pandas_dataframes_with_dagster_types.mdx

```diff
@@ -0,0 +1,240 @@
+
+Notice also that we provide a `metadata` dictionary to the `TypeCheck` constructor when validation fails. This can hold arbitrary information that is visible in [Dagit](/concepts/dagit) and can be useful in debugging. Here we've only provided a count `num_violations` of the schema violations (`SchemaErrors.failure_cases` is a dataframe tabulating each violation), but we could easily include richer information to annotate more complex validations.
+
+## Apply our Dagster Type to our Job
+
+Now that we can turn Pandera schemas into Dagster types, we only need to tweak our job definition a little bit to integrate Pandera validation.
+
+```python file=/guides/validating_pandas_dataframes_with_dagster_types/job_2.py endbefore=execution
+from dagster import AssetMaterialization, In, job, op, Out
+import numpy as np
+import pandas as pd
+import pandera as pa
+import holoviews as hv
+
+from .factory import pandera_schema_to_dagster_type
+
+MIN_DATE = pd.Timestamp('2021-10-01')
+
+schema = pa.DataFrameSchema(
+    columns = {
+        "bike_id": pa.Column(int, checks=pa.Check.ge(0)),  # ge: greater than or equal to
+        "start_time": pa.Column(pd.Timestamp, checks=pa.Check.ge(MIN_DATE)),
+        "end_time": pa.Column(pd.Timestamp, checks=pa.Check.ge(MIN_DATE)),
+    },
+)
+
+# This is a Dagster type that wraps the schema
+TripsDataFrame = pandera_schema_to_dagster_type(
+    schema, 'TripsDataFrame', 'DataFrame type for e-bike trips.'
+)
+
+
+# We've added a Dagster type for this op's output
+@op(out=Out(TripsDataFrame), config_schema={'clean': bool})
+def load_trips(context):
+    df = pd.read_csv(
+        "./ebike_trips.csv",
+        parse_dates=["start_time", "end_time"],
+    )
+    if context.op_config['clean']:
+        df = df[pd.notna(df.end_time)]
+    return df
+
+# We've added a Dagster type for this op's input
+@op(ins={'trips': In(TripsDataFrame)})
+def generate_plots(trips):
+    minute_lengths = [ x.total_seconds() / 60 for x in trips.end_time - trips.start_time ]
+    edges, freqs = np.histogram(minute_lengths, 15)
+    hist = hv.Histogram((freqs, edges)).opts(width=600, xlabel='Minutes')
+    hv.save(hist, 'trip_length_hist.png', fmt='png')
+    yield AssetMaterialization(
+        asset_key="trip_dist_plot",
+        description="Distribution of trip lengths."
+    )
+    yield Output(None)
+
+@job
+def generate_trip_plots():
+    generate_plots(load_trips)
+```
+
+The above job definition makes two modifications to the previous one. First, we've added `ins`/`out` arguments to the `op` decorators. We specify that `load_trips` must return a `TripsDataframe` as output, and `generate_plots` must take a `TripsDataframe` as input. Now when we execute the `generate_trip_plots` job, Dagster will automatically apply the `TripsDataframe` logic to our dataframe before it enters `generate_plots`.
```

Comment:
Nit: TripsDataFrame

Reply:
instead of "apply the logic", maybe "apply the validation function... and fail if it fails"